helm/templates/loki/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: {{ template "loki.fullname" . }}
+  labels:
+    app: {{ template "loki.name" . }}
+    chart: {{ template "loki.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+{{- with .Values.loki.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+  replicas: {{ .Values.loki.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "loki.name" . }}
+      release: {{ .Release.Name }}
+  strategy:
+    type: {{ .Values.loki.deploymentStrategy }}
+  {{- if ne .Values.loki.deploymentStrategy "RollingUpdate" }}
+    rollingUpdate: null
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "loki.name" . }}
+        release: {{ .Release.Name }}
+{{- with .Values.loki.podAnnotations }}
+      annotations:
+{{ toYaml . | indent 8 }}
+{{- end }}
+    spec:
+      serviceAccountName: {{ template "loki.serviceAccountName" . }}
+{{- if .Values.loki.priorityClassName }}
+      priorityClassName: {{ .Values.loki.priorityClassName }}
+{{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.loki.image.repository }}:{{ .Values.loki.image.tag }}"
+          imagePullPolicy: {{ .Values.loki.image.pullPolicy }}
+          args: 
+            - "-config.file=/etc/loki/loki.yaml"
+          volumeMounts:
+            - name: config
+              mountPath: /etc/loki
+              # subPath: loki.yaml
+          ports:
+            - name: service
+              containerPort: {{ .Values.loki.service.port }}
+              protocol: TCP
```

Comment:
I don't understand what this port is for?

Reply:
Rather, you've told Loki to listen on {{ .Values.loki.service.port }} in the config, so this is accurate - but you've also added a loki port below for 3100.  Why do you need both?

Reply:
I cleared the service port